third_party/move/move-compiler-v2/tests/checking/receiver/calls_index.move

```diff
@@ -0,0 +1,61 @@
+module 0x42::m {
+
+    struct S has key, drop { t: T }
+
+    struct T has key, store, drop {
+        w: W
+    }
+
+    struct W has key, store, drop {
+        x: u64
+    }
+
+    fun merge(self: &mut W, s: W) {
+        self.x += s.x;
+    }
+
+    fun foo(account: address, w: W) acquires S {
+        S[account].t.w.merge(w)
+    }
+
+    fun foo_2(account: address, w: W) acquires W {
+        W[account].merge(w)
```

Comment:
Could we add the following test case (or similar) that tests:
1. Non-mut ref receiver over resource indexing
2. Resource indexing on a generic type that goes through the newly added translation.

```
    struct Wrapper<T> has drop, key, store, copy {
        inner: T
    }

    fun unwrap<T>(self: &Wrapper<T>): T {
        self.inner
    }

    fun dispatch<T>(account: address): T acquires Wrapper {
        Wrapper<T>[account].unwrap()
    }
```

Reply:
done